_pytest/python.py

```diff
@@ -2522,13 +2527,52 @@ def execute(self, request):
                     fixturefunc = fixturefunc.__get__(request.instance)
 
         try:
-            result = call_fixture_func(fixturefunc, request, kwargs)
+            config = request.config
+            if config.option.setupplan:
+                result = None
+            else:
+                result = call_fixture_func(fixturefunc, request, kwargs)
+            if config.option.setuponly or config.option.setupplan:
+                # We want to access the params of ids if they exist also in during
+                # the finish() method.
+                if hasattr(request, 'param'):
+                    if self.ids:
+                        ind = self.params.index(request.param)
+                        self.cached_param = self.ids[ind]
+                    else:
+                        self.cached_param = request.param
+                self._log_fixture_stack('SETUP')
         except Exception:
             self.cached_result = (None, my_cache_key, sys.exc_info())
             raise
         self.cached_result = (result, my_cache_key, None)
         return result
 
+    def _log_fixture_stack(self, what):
+        config = self._fixturemanager.config
+        capman = config.pluginmanager.getplugin('capturemanager')
+        if capman:
+            capman.suspendcapture()
```

Comment:
You're right, we didn't realise that `.suspendcapture()` + `.resumecapture()` loses the output unless explicitly passed through. Fixed.